src/perl/lib/WTSI/NPG/Genotyping/VCF/Slurper.pm

```diff
@@ -0,0 +1,145 @@
+use utf8;
+
+package WTSI::NPG::Genotyping::VCF::Slurper;
+
+use Moose;
+
+use File::Temp qw /tempdir/;
+use WTSI::NPG::iRODS;
+use WTSI::NPG::iRODS::DataObject;
+use WTSI::NPG::Genotyping::VCF::DataRowParser;
+use WTSI::NPG::Genotyping::VCF::HeaderParser;
+
+with 'WTSI::DNAP::Utilities::Loggable';
+
+has 'input_path'  =>
+   (is            => 'ro',
+    isa           => 'Str',
+    required      => 1,
+    documentation => "Input path in iRODS or local filesystem,".
+                     " or '-' for STDIN.",
+   );
+
+has 'irods'       =>
+   (is            => 'ro',
+    isa           => 'WTSI::NPG::iRODS',
+    documentation => '(Optional) iRODS instance from which to read data. '.
+                     'If not given, input (if any) is assumed to be a '.
+                     'path on the local filesystem.',
+   );
+
```

Comment:
Hi again 

On reflection, I decided it was easier and less confusing to just make
the input_filehandle attribute private, so I've done that (insofar as
Moose allows). Setting 'init_arg => undef' ensures it can't be supplied
at object creation time. 

I've also made changes for compatibility with vcf_validator; it now says
the VCF files produced by my code are valid. 

Both sets of changes have been added to the pull request. 

Have a good evening and I'll see you tomorrow, 

Cheers 

Iain 

On 2015-07-01 17:19, Keith James wrote: 

> In src/perl/lib/WTSI/NPG/Genotyping/VCF/Slurper.pm [1]: 
> 
>> +has 'input_path' =>
>> + (is => 'ro',
>> + isa => 'Str',
>> + required => 1,
>> + documentation => "Input path in iRODS or local filesystem,".
>> + " or '-' for STDIN.",
>> + );
>> +
>> +has 'irods' =>
>> + (is => 'ro',
>> + isa => 'WTSI::NPG::iRODS',
>> + documentation => '(Optional) iRODS instance from which to read data. '.
>> + 'If not given, input (if any) is assumed to be a '.
>> + 'path on the local filesystem.',
>> + );
>> +
> 
> What is expected to happen when I pass an 'input_filehandle' initarg to the constructor? 
> 
> --
> Reply to this email directly or view it on GitHub [2]. 
> 
> --> --> -->

-- 
Iain Bancarz PhD
Senior Software Developer
Sequence Informatics :: Team 117
Wellcome Trust Sanger Institute
Tel. +44 (0)1223 834244 extension 8211
 

Links:
------
[1] https://github.com/wtsi-npg/genotyping/pull/358#discussion_r33695578
[2] https://github.com/wtsi-npg/genotyping/pull/358/files#r33695578



-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.